Networking automation

TregubovYury opened the pull request:
Author: Tregubov Yury
Added os_interface the class to be used for communication with
deployed stack over neutron nova and the rest clients

Commit:
Networking automation

Author: Tregubov Yury
Added os_interface the class to be used for communication with
deployed stack over neutron nova and the rest clients

gdyuldin commented on tools/settings.py:
```diff
@@ -22,6 +22,17 @@
+    return _boolean_states.get(value.lower(), default)
+
+PATH_TO_CERT = os.environ.get('PATH_TO_CERT', os.path.join(
+    SSL_CERTS_DIR, 'ca.crt'))
```
SSL_CERTS_DIR is not defined here

TregubovYury replied:
Hm... flake did not caught it. Tnx will fix that.

gdyuldin commented on test_enviroment/os_interface.py:
```diff
@@ -0,0 +1,204 @@
+            raise
+        return keystone
+
+    def cleanup_network(self, networks_to_skip=[]):
```
May be would be better to split this method to some parts, and call them here. Also, perhaps 'cleanup' or 'cleanup_env' will be a better name for this method.

TregubovYury replied:
This method intends to clean up only the network configs. By the env cleanup I think at least nodes revert should be meant. 
Also I think that we can just add all needed helpers for the nova neutron etc in this class.
So all helpers for the clients are located in one file.

TregubovYury replied:
And regarding the splitting to the parts... I don't think so.
The only way you can delete the objects form the bottom of the function is to delete the objects from the beginning. And the deletion itself takes only one function call.

Commit:
Network automation

Author: Tregubov Yury
Updated some comments from Georgy

Commit:
Networking automation

Added a class VirtualEnviroment with an example how to revert the
snapshot with the devops only

TimurNurlygayanov commented on enviroment/os_actions.py:
```diff
@@ -0,0 +1,204 @@
+                    logger.info('The Security Group {} is not deletable'
+                                 .format(sg))
+
+        # After some experiments the followin sequence for deleteion was found
```
followin  > following

deleteion > deletion

TimurNurlygayanov commented on enviroment/os_actions.py:
```diff
@@ -0,0 +1,204 @@
+                                 .format(sg))
+
+        # After some experiments the followin sequence for deleteion was found
+        # roter_interface and ports -> subnets -> routers -> nets
```
roter_interface  > router_interface

TimurNurlygayanov commented on enviroment/os_actions.py:
```diff
@@ -0,0 +1,204 @@
+
+        # After some experiments the followin sequence for deleteion was found
+        # roter_interface and ports -> subnets -> routers -> nets
+        # Delete router interafce and ports
```
interafce  > interaface

TimurNurlygayanov commented on enviroment/os_actions.py:
```diff
@@ -0,0 +1,204 @@
+        # Delete router interafce and ports
+        # TBD some ports are still kept after the cleanup.
+        # Need to find why and delete them as well
+        # But it does not fail the executoin so far.
```
executoin  > execution

TregubovYury replied:
Huh... need to install some plugin for spell-check in vim already...

TimurNurlygayanov commented on enviroment/os_actions.py:
```diff
@@ -0,0 +1,204 @@
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""Doc."""
```
Why we need this string here?

Commit:
Networking automation

Author: Tregubov Yury
Fixed some spelling mistakes in comments

Commit:
Networking automation

Author: Tregubov Yury
Added get_admin_node_ip in virt_env.py

TimurNurlygayanov commented on enviroment/virt_env.py:
```diff
@@ -0,0 +1,84 @@
+from tools.setting import logger
+
+
+class VrtualEnviroment(object):
```
Vrtual > Virtual

TimurNurlygayanov commented on enviroment/virt_env.py:
```diff
@@ -0,0 +1,84 @@
+        Will return empty string if admin node was not found in env.
+        """
+        admin_ip = ''
+        env = VrtualEnviroment.get_env(env_name)
```
Vrtual - the same here

Commit:
Netwrokig automation

Author: Tregubov Yury
Fixed import of settings and some spelling mistakes

Commit:
Networking automation

Author: Tregubov Yury

Did some changes in get_env and revert_snapshot methods
